### PR TITLE
Tila 2840 UI refactor rest to gql

### PR DESCRIPTION
**Pull request by joonatank**
Replace all REST api with GQL in ui (except for calendar url).
Update the application gql api to match the new one in core.

Refs: TILA-2840, TILA-2844

### CODE REVIEW

File: apps/ui/hooks/useApplicationUpdate.tsx [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+export const useApplicationUpdate = () => {
+  const [mutate, { error, loading: isLoading }] = useMutation<Mutation, MutationUpdateApplicationArgs>(
+    UPDATE_APPLICATION_MUTATION,
+    {
+      /*
+      onError: (e) => {
+        console.warn("update application mutation failed: ", e);
+        setError(`${t("application:error.saveFailed")}`);
+      },
+      */
```
Comment (joonatank) [inline apps/ui/hooks/useApplicationUpdate.tsx:18]: Remove

File: apps/ui/hooks/useApplicationUpdate.tsx [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+        },
+      });
+      // TODO cleanup the error handling
+      // TODO translate errors
+      const { data, errors } = response;
+      const { errors: mutErrors, pk } = data?.updateApplication ?? {};
+      if (errors != null) {
+        console.error("Error saving application: ", errors);
+        // TOOD display to the user
+        // setError("Error saving application");
```
Comment (joonatank) [inline apps/ui/hooks/useApplicationUpdate.tsx:36]: Clean

File: apps/ui/pages/application/[...params].tsx [hunk 3]
```diff
@@ -47,281 +42,262 @@ export const getServerSideProps: GetServerSideProps = async (ctx) => {
...
+
+  if (id == null) {
+    return <Error statusCode={404} />;
+  }
+  if (isLoading) {
+    return <CenterSpinner />;
+  }
+  if (application == null || applicationRound == null) {
+    return <Error statusCode={404} />;
+  }
```
Comment (joonatank) [inline apps/ui/pages/application/[...params].tsx:292]: Slug check?

File: apps/ui/pages/application/[id]/page3.tsx [hunk 4]
```diff
@@ -0,0 +1,269 @@
...
+  return {
+    name: org?.name || undefined,
+    identifier: org?.identifier || undefined,
+    address: transformAddress(org?.address),
+  };
+};
+
+/// TODO this isn't validated when saving
+/// make a separate schema for it? and maybe even a separate form?
+const Page3 = (): JSX.Element | null => {
```
Comment (joonatank) [inline apps/ui/pages/application/[id]/page3.tsx:75]: Move down after utility functions.

**Review by vincit-matu — APPROVED** [review]
Fix commented stuff, but otherwise lgtm

**Pull request merged**